### DISCUSSION TITLE
Questions on building on Windows from a git clone

### DISCUSSION THREAD

**Discussion by clydegerber** [DOCGroup/ACE_TAO#1837] [General]
I'm trying to follow the instructions in [Building from git](https://www.dre.vanderbilt.edu/~schmidt/DOC_ROOT/ACE/ACE-INSTALL.html#power) and am running into a problem executing this command:

$ACE_ROOT/bin/mwc.pl -type vs2019 ACE.mwc

The error I'm getting is: 

ERROR: Unable to find the MPC modules in C:/<my folder>/ACE_TAO/ACE/MPC.
You can set the MPC_ROOT environment variable to the location of MPC.

I don't see an 'MPC' folder under "ACE_TAO/ACE" in the cloned repo (or in the origin). I do see that folder when I download the source zip file from https://download.dre.vanderbilt.edu/. If I copy that MPC folder to the ACE_TAO/ACE folder in the cloned repo the above command works.

Should there be an MPC folder in the git repo under ACE_TAO/ACE?

Thanks!

**Comment by iguessthislldo** [comment]
Yes, the release archives include MPC, but when building from the raw source you have to supply MPC yourself as it's a [separate repository](https://github.com/DOCGroup/MPC). The instructions seem to assume that and could be changed to be more clear. You could move the MPC from the release archive like you mentioned, but it's probably a better idea to clone MPC into `ACE` or somewhere else if you set `MPC_ROOT` like the error message mentions.

**Reply by clydegerber** [reply]
Thanks for the info.

**Comment by mitza-oci** [comment]
Also, this repository contains a GitHub Actions YAML file that may be helpful in that it shows each step used to build on Windows/MSVC.

https://github.com/DOCGroup/ACE_TAO/actions/runs/2325915632/workflow
